cli/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -662,36 +746,46 @@ protected void updateExistingVersions(UrlEdition edition) {
   private void updateExistingVersion(String edition, String version, UrlVersion urlVersion, StatusJson statusJson,
       UrlStatusFile urlStatusFile) {
 
-    boolean modified = false;
     String toolWithEdition = getToolWithEdition(edition);
     Instant now = Instant.now();
-    for (UrlFile<?> child : urlVersion.getChildren()) {
-      if (child instanceof UrlDownloadFile) {
-        Set<String> urls = ((UrlDownloadFile) child).getUrls();
+    // since Java collections do not support modification while iterating, we need to create a copy
+    Collection<UrlFile<?>> urlFiles = new ArrayList<>(urlVersion.getChildren());
+    for (UrlFile<?> child : urlFiles) {
+      if (child instanceof UrlDownloadFile urlDownloadFile) {
+        // since Java collections do not support modification while iterating, we need to create a copy
+        Set<String> urls = new HashSet<>(urlDownloadFile.getUrls());
         for (String url : urls) {
-          if (shouldVerifyDownloadUrl(version, statusJson, toolWithEdition, now)) {
+          if (shouldVerifyDownloadUrl(url, statusJson, toolWithEdition, now)) {
```

Comment:
very evil bug: first argument is `url` but `version` was given as 1st arg before this change. 👿